docs/en/modules/perf.md

```diff
@@ -0,0 +1,61 @@
+# perf Module
+
+This module provides simple performance measurement for an application.
+It samples the program counter roughly every 50 microseconds and builds
+a histogram of the values that it finds. Since there is only a small amount
+of memory to store the histogram, the user must specify which area of code
+is of interest. This can then be repeated with different areas and at different
+resolutions to get as much information as required.
+
+## perf.start()
+Starts a performance monitoring session.
+
+####Syntax
+`perf.start(start, end, nbins[, offset])`
+
+####Parameters
+- `start` The lowest PC address for the histogram
+- `end` The highest address for the histogram
+- `nbins` The number of bins in the histogram. Keep this reasonable otherwise 
+you will run out of memory.
+- `offset` This is an optional field to specify the offset of the saved PC value
+on the interrupt stack. It appears that 16 is the correct value.
+
+####Returns
+Nothing
+
+## perf.stop()
+
+Terminates a performance monitoring session and returns the histogram.
+
+####Syntax
+`total, outside, histogram = perf.stop()`
+
+####Returns
+- `total` The total number of samples captured in this run
+- `outside` The number of samples that were outside the histogram range
+- `histogram` The histogram represented as a table indexed by address where the value is the number of samples. The address is the lowest address for the bin.
+
+### Example
+
+    perf.start(0x40200000, 0x40300000, 1024)
+
```

Comment:
You have to use the mapfile that is generated by the build. If you want to find a hotspot, you can try a large range (say from 0x40000000 to 0x40400000) and then do the test and look at the results. This will show you (roughly) where the time is being spent and then you can run again just over that range...